iq_learn/train_iq.py

```diff
@@ -181,35 +181,23 @@ def main(cfg: DictConfig):
                 agent.batch_size, agent.device
             )
             expert_obs, _, expert_action, __, ___, expert_cond_detached, true_traj_idx = expert_batch
-            unique_idx = []
-            expert_obs_unique = []
-            expert_action_unique = []
-            for i, traj_idx in enumerate(true_traj_idx):
-                if traj_idx not in unique_idx:
-                    unique_idx.append(traj_idx)
-                    expert_obs_unique.append(expert_obs[i])
-                    expert_action_unique.append(expert_action[i])
-            expert_obs_unique = torch.stack(expert_obs_unique, dim=0)   
-            expert_action_unique = torch.stack(expert_action_unique, dim=0)
             emb_list = get_new_cond(
                 encoder, 
                 hydra.utils.to_absolute_path(f'experts/{args.env.demo}'), 
                 hydra.utils.to_absolute_path('cond/temp_cond.pkl'), 
                 device,
-                unique_idx)
+                true_traj_idx)
             dist_params_list = emb_list["dist_params"]
-            dist_params_list = [dist_params_list[i] for i in unique_idx]
             mu = [dist_params[0] for dist_params in dist_params_list]     
             log_var = [dist_params[1] for dist_params in dist_params_list]
             mu = torch.stack(mu, dim=0)
             log_var = torch.stack(log_var, dim=0)
            
             expert_cond_list = emb_list["emb"]
-            expert_cond_list = [expert_cond_list[i] for i in unique_idx]
             expert_cond = torch.stack(expert_cond_list, dim=0)
```

Comment:
The cond from encoder goes here. 
1. cond is forwarded by encoder at each training step
2. cond is passed to agent.bc_update() for forwarding through the actor
3. agent.bc_update() calculate loss afterwards